// @makerdao/ui-components-header/src/MobileHeaderDropdown.jsx
import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";
import { Flex, Box } from "@makerdao/ui-components-core";
import downCaret from "./assets/downCaret.inline.svg";

const MobileDropdownHeader = styled(Flex)`
  color: ${props => props.theme.header.linkHeaderColor};
  cursor: pointer;
  font-size: 1.95rem;

  &:after {
    transition: transform 0.2s;
    content: "";
    mask: url(${downCaret}) center center no-repeat;
    background-color: ${props => props.theme.header.linkHeaderColor};
    width: 2.8rem;
    height: 1.6rem;
    margin-right: 0.6rem;
    ${props => props.open && `transform: rotate(180deg);`}
  }
`;

const MobileDropdownContainer = styled(Box)`
  &:not(:last-child) {
    border-bottom: 1px solid rgba(126, 127, 154, 0.2);
  }
`;

const MobileDropdownItemsContainer = styled.div`
  opacity: 0;
  height: auto;
  max-height: 0px;
  transform-origin: top;
  transition: all 0.25s ease-in-out;
  pointer-events: none;

  ${props =>
    props.open &&
    `
    pointer-events: unset;
    max-height: 500px;
    opacity: 1;
  `}
`;

class MobileHeaderDropdown extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      open: false,
    };
  }

  toggleDropdown() {
    this.setState({
      open: !this.state.open,
    });
  }

  render() {
    return (
      <MobileDropdownContainer pt="1.4rem" pb="1rem">
        <MobileDropdownHeader
          pb="0.7rem"
          justifyContent="space-between"
          alignItems="center"
          open={this.state.open}
          onClick={this.toggleDropdown.bind(this)}
        >
          {this.props.title}
        </MobileDropdownHeader>
        <MobileDropdownItemsContainer open={this.state.open}>
          {this.props.children}
        </MobileDropdownItemsContainer>
      </MobileDropdownContainer>
    );
  }
}

MobileHeaderDropdown.propTypes = {
  title: PropTypes.string,
  children: PropTypes.node,
};

export default MobileHeaderDropdown;
