// @makerdao/design-system-constants/src/colors.js
const baseColors = {
  teal: {
    "100": "#E7FCFA",
    "200": "#B6EDE7",
    "300": "#6EDBD0",
    "400": "#2DC1B1",
    "500": "#1AAB9B",
    "600": "#139D8D",
    "700": "#098C7D",
    "800": "#027265",
    "900": "#06554C",
  },
  yellow: {
    "100": "#FFF9ED",
    "200": "#FFF1CF",
    "300": "#FEDB88",
    "400": "#FBCC5F",
    "500": "#FDC134",
    "600": "#F9A606",
    "700": "#F08B04",
    "800": "#E67002",
    "900": "#DC5D00",
  },
  orange: {
    "100": "#FDEDE8",
    "200": "#FBE1D9",
    "300": "#FAB6A1",
    "400": "#F99374",
    "500": "#F77249",
    "600": "#F75524",
    "700": "#EB4714",
    "800": "#CB3A0D",
    "900": "#A83815",
  },
  slate: {
    "100": "#EDF5F9",
    "200": "#D1DEE6",
    "300": "#9FAFB9",
    "400": "#708390",
    "500": "#546978",
    "600": "#405361",
    "700": "#31424E",
    "800": "#1E2C37",
    "900": "#10191F",
  },
  grey: {
    "100": "#F5F5F5",
    "200": "#EBEBEB",
    "300": "#D6D6D6",
    "400": "#C2C2C2",
    "500": "#ADADAD",
  },
  coolGrey: {
    "100": "#F6F8F9",
    "200": "#ECF1F3",
    "300": "#E7EDF0",
    "400": "#DDE5E7",
    "500": "#D8E0E3",
  },
};

const deprecatingColors = {
  linkOrange: "#F35833",
  heading: "#231536",
  text: "#48495F",
  linkBlue: "#447AFB",
  backgroundGrey: baseColors.coolGrey["100"],
};

const colors = {
  ...baseColors,
  blue: deprecatingColors.linkBlue,
  lightBlue: "#E9EFFF",
  black: "#000000",
  white: "#FFFFFF",
  makerTeal: baseColors.teal["500"],
  makerOrange: baseColors.orange["500"],
  daiYellow: baseColors.yellow["500"],
  darkPurple: deprecatingColors.heading,
  darkLavender: deprecatingColors.text,
  steel: baseColors.slate["400"],
  steelLight: baseColors.slate["300"],
  borderGrey: "#D4D9E1",
  lightGrey: deprecatingColors.backgroundGrey,
};

export default colors;
