// @makerdao/design-system-constants/src/typography.js
import colors from "./colors";

export const fontWeights = {
  thin: "100",
  extralight: "200",
  light: "300",
  normal: "400",
  medium: "500",
  semibold: "600",
  bold: "700",
  extrabold: "800",
  black: "900",
};

export const fonts = {
  sans: `-apple-system, system-ui, BlinkMacSystemFont, 'SF Pro Text', 'Segoe UI', Roboto, Helvetica, Arial, sans-serif;`,
};

export const fontSizes = {
  xs: "1.1rem",
  s: "1.4rem",
  m: "1.5rem",
  l: "1.6rem",
  xl: "2rem",
  "2xl": "2.6rem",
  "3xl": "3.2rem",
  "4xl": "4.1rem",
};

fontSizes.base = fontSizes.m;

// ask design
export const lineHeights = {
  none: "1",
  tight: "1.2",
  normal: "1.5",
  loose: "2",
};

export const letterSpacings = {
  tight: "-0.05",
  normal: "0",
  wide: "0.05em",
};

export const typography = {
  h1: {
    fontSize: fontSizes["4xl"],
    lineHeight: lineHeights.none,
    fontFamily: fonts.sans,
    fontWeight: fontWeights.normal,
    color: colors.darkPurple,
    letterSpacing: letterSpacings.normal,
  },
  h2: {
    fontSize: fontSizes["3xl"],
    lineHeight: lineHeights.none,
    fontFamily: fonts.sans,
    fontWeight: fontWeights.medium,
    color: colors.darkPurple,
    letterSpacing: letterSpacings.normal,
  },
  h3: {
    fontSize: fontSizes["2xl"],
    lineHeight: lineHeights.none,
    fontFamily: fonts.sans,
    fontWeight: fontWeights.medium,
    color: colors.darkPurple,
    letterSpacing: letterSpacings.normal,
  },
  h4: {
    fontSize: fontSizes.xl,
    lineHeight: lineHeights.none,
    fontFamily: fonts.sans,
    fontWeight: fontWeights.medium,
    color: colors.darkPurple,
    letterSpacing: letterSpacings.normal,
  },
  h5: {
    fontSize: fontSizes.l,
    lineHeight: lineHeights.none,
    fontFamily: fonts.sans,
    fontWeight: fontWeights.medium,
    color: colors.darkPurple,
    letterSpacing: letterSpacings.normal,
  },
  body: {
    fontSize: fontSizes.m,
    lineHeight: lineHeights.normal,
    fontFamily: fonts.sans,
    color: colors.darkLavender,
    letterSpacing: letterSpacings.normal,
  },
  subheading: {
    fontSize: fontSizes.xs,
    textTransform: "uppercase",
    letterSpacing: letterSpacings.wide,
    lineHeight: lineHeights.none,
    fontWeight: fontWeights.bold,
    fontFamily: fonts.sans,
    color: colors.steel,
  },
  caption: {
    fontSize: fontSizes.s,
    fontFamily: fonts.sans,
    lineHeight: lineHeights.none,
    letterSpacing: letterSpacings.normal,
    color: colors.steel,
  },
};
