// @makerdao/ui-components-onboarding/src/OnboardingMenu.jsx
import React from "react";
import styled from "styled-components";
import PropTypes from "prop-types";

import { Grid, Flex, Box } from "@makerdao/ui-components-core";
import closeIcon from "./assets/close.inline.svg";
import closeIconFilled from "./assets/close-filled.inline.svg";

const HeaderButtons = styled.div`
  cursor: pointer;
`;

const HoverIcon = styled.div`
  height: ${props => props.height};
  width: ${props => props.width};
  position: relative;

  *:hover > &:before {
    opacity: 0;
  }

  *:hover > &:after {
    opacity: 1;
  }

  &:before,
  &:after {
    content: "";
    height: ${props => props.height};
    width: ${props => props.width};
    position: absolute;
    transition: opacity 0.2s;
  }

  &:before {
    content: url(${props => props.startImg});
  }

  &:after {
    content: url(${props => props.hoverImg});
    opacity: 0;
  }
`;

const OnboardingMenu = props => {
  return (
    <Grid
      gridTemplateColumns="auto auto"
      gridColumnGap="xl"
      justifyContent="flex-end"
    >
      <HeaderButtons onClick={props.onClose}>
        <Flex alignItems="center" color="#8D8EA7">
          <HoverIcon
            height="2.4rem"
            width="2.4rem"
            color="#8D8EA7"
            startImg={closeIcon}
            hoverImg={closeIconFilled}
          />
          <Box ml="xs">Save and Exit</Box>
        </Flex>
      </HeaderButtons>
    </Grid>
  );
};

OnboardingMenu.propTypes = {
  onClose: PropTypes.func.isRequired,
};

export default OnboardingMenu;
