// @makerdao/ui-components-footer/src/LangDropdown.jsx
import React from "react";
import styled, { withTheme } from "styled-components";

import {
  Dropdown,
  DefaultDropdown,
  Box,
  Flex,
} from "@makerdao/ui-components-core";
import UpCaret from "./assets/upCaret.svg";
import Globe from "./assets/globe.svg";
import { arrayOf, shape, string, func } from "prop-types";

const Trigger = withTheme(({ children, ...props }) => (
  <Flex alignItems="center">
    <Globe />
    <Box ml="0.9rem" mr="0.6rem" color={props.theme.colors.text}>
      {children}
    </Box>
    <UpCaret />
  </Flex>
));

const Item = styled.div`
  color: ${props => props.theme.footer.linkColor};
  line-height: 2.4rem;
  transition: color 0.2s ease-in-out;
  text-decoration: none;
  font-size: 1.6rem;
  cursor: pointer;

  &:hover {
    color: black;
  }
`;

const LangDropdown = ({
  languages,
  selectedValue,
  defaultText,
  onSelect,
  ...props
}) => {
  const langText = languages.find(l => l.value === selectedValue).text;
  return (
    <Dropdown
      trigger={<Trigger>{langText || defaultText}</Trigger>}
      placement="top-start"
      offset="0, 10px"
      hitBoxMargin="10px"
      {...props}
    >
      <DefaultDropdown>
        {languages.map(({ value, text }) => (
          <Item key={value} onClick={e => onSelect(e, value, text)}>
            {text}
          </Item>
        ))}
      </DefaultDropdown>
    </Dropdown>
  );
};

LangDropdown.propTypes = {
  /** Array of languages in the form {value, text}, where value is the locale, and text is the displayed language. */
  languages: arrayOf(
    shape({
      value: string.isRequired,
      text: string.isRequired,
    })
  ).isRequired,

  /** The currently selected locale **/
  selectedValue: string,

  /** The text to display when no language is selected */
  defaultText: string,

  /** Callback when a language is selected. Callback is called with (e, value, text) **/
  onSelect: func,
};

LangDropdown.defaultProps = {
  defaultText: "Select",
};

export default LangDropdown;
